torchgeo/datasets/utils.py

```diff
@@ -847,3 +851,20 @@ def array_to_tensor(array: np.typing.NDArray[Any]) -> Tensor:
     elif array.dtype == np.uint32:
         array = array.astype(np.int64)
     return torch.tensor(array)
+
+
+def valid_data_footprint_from_datasource(
+    src: rasterio.io.DatasetReader, destination_crs: pyproj.crs.crs.CRS
+) -> Polygon:
+    # Read valid/nodata-mask
+    mask = src.read_masks()
+    # Close holes
+    sieved_mask = sieve(mask, 500)
+    # extract polygon for valid data values
+    geom = next(g for g, v in shapes(sieved_mask, transform=src.transform) if v > 0)
```

Comment:
instead of the first polygon, maybe we should go for the bigger polygon? Or if we aim to handle everything, maybe it should be a join between the geometries

Reply:
I am now pushing a suggestion where we keep all polygons (that are not nodata) and make a multipolygon. This will then support "complex" footprints, or disjunct mosaics

Reply:
> I am now pushing a suggestion where we keep all polygons (that are not nodata) and make a multipolygon. This will then support "complex" footprints, or disjunct mosaics

Nice, I would give it as a suggestion because I see some cases like this on a daily basis 😄